src/ActionList/Heading.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {forwardRef} from 'react'
+import {BetterSystemStyleObject, SxProp, merge} from '../sx'
+import {defaultSxProp} from '../utils/defaultSxProp'
+import {useRefObjectAsForwardedRef} from '../hooks'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+import {default as HeadingComponent} from '../Heading'
+import {ListContext} from './List'
+
+export type ActionListHeadingProps = {
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & SxProp
+
+export const Heading = forwardRef(({as: Component = 'h3', children, sx = defaultSxProp, ...props}, forwardedRef) => {
```

Comment:
Does `role="list"` vs `role="listbox"` matter for `ActionList.Heading` implementation? 

From what I understand on [this comment](https://github.com/github/primer/pull/2484#issuecomment-1658368122), it only impacts the group headings as `ActionList.Heading` is rendered outside of the list anyway?

Reply:
> it only impacts the group headings as ActionList.Heading is rendered outside of the list anyway?

yep, that is correct :)

Reply:
Ooo great point! I'll ask Kendall

Reply:
We get this point clarified: 
> Semantic `<menu>`s don't have a place for headers within them, they should be aria-labelledby the menu button's name. 

by Chelsea ([reference](https://github.com/github/accessibility-audits/issues/2939#issuecomment-1675375647))